org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/migration/AbstractRemoveDoEntityContributionValueMigrationHandlerTest.java

```diff
@@ -0,0 +1,276 @@
+import java.util.stream.Collectors;
+
+import org.eclipse.scout.rt.dataobject.DoEntity;
+import org.eclipse.scout.rt.dataobject.IDataObjectMapper;
+import org.eclipse.scout.rt.dataobject.IDoEntity;
+import org.eclipse.scout.rt.dataobject.ITypeVersion;
+import org.eclipse.scout.rt.dataobject.fixture.FirstSimpleContributionFixtureDo;
+import org.eclipse.scout.rt.dataobject.fixture.SecondSimpleContributionFixtureDo;
+import org.eclipse.scout.rt.dataobject.fixture.SimpleFixtureDo;
+import org.eclipse.scout.rt.dataobject.migration.DataObjectMigrator.DataObjectMigratorResult;
+import org.eclipse.scout.rt.dataobject.migration.fixture.version.AlfaFixtureTypeVersions.AlfaFixture_1;
+import org.eclipse.scout.rt.dataobject.migration.fixture.version.AlfaFixtureTypeVersions.AlfaFixture_2;
+import org.eclipse.scout.rt.platform.BEANS;
+import org.eclipse.scout.rt.platform.util.CollectionUtility;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Tests {@link AbstractRemoveDoEntityContributionValueMigrationHandler}
+ */
+public class AbstractRemoveDoEntityContributionValueMigrationHandlerTest {
+
+  protected static IDataObjectMapper s_dataObjectMapper;
+
+  @BeforeClass
+  public static void beforeClass() {
+    s_dataObjectMapper = BEANS.get(IDataObjectMapper.class);
+  }
+
+  /**
+   * Tests that data object is not changed if there is no contribution.
+   */
+  @Test
+  public void testMigrationOfDoEntityWithoutContribution() {
+    SimpleFixtureDo fixture = createSimpleFixtureDo();
+
+    String fixtureJson = s_dataObjectMapper.writeValue(fixture);
+    DataObjectMigrationContext ctx = BEANS.get(DataObjectMigrationContext.class).putGlobal(BEANS.get(DoValueMigrationIdsContextData.class).withAppliedValueMigrationIds(getAppliedValueMigrations()));
+    DataObjectMigratorResult<SimpleFixtureDo> result = BEANS.get(DataObjectMigrator.class).migrateDataObject(ctx, fixtureJson, SimpleFixtureDo.class);
+    assertFalse(result.isChanged());
+    assertEquals(fixture, result.getDataObject());
+  }
+
+  /**
+   * Tests that data object is not changed if there is no matching contribution.
+   */
+  @Test
+  public void testMigrationOfDoEntityWithDifferentContribution() {
+    SimpleFixtureDo fixture = createSimpleFixtureDo();
+    fixture.contribution(SecondSimpleContributionFixtureDo.class)
+        .withSecondValue("mySecondValue");
+
+    String fixtureJson = s_dataObjectMapper.writeValue(fixture);
+    DataObjectMigrationContext ctx = BEANS.get(DataObjectMigrationContext.class).putGlobal(BEANS.get(DoValueMigrationIdsContextData.class).withAppliedValueMigrationIds(getAppliedValueMigrations()));
+    DataObjectMigratorResult<SimpleFixtureDo> result = BEANS.get(DataObjectMigrator.class).migrateDataObject(ctx, fixtureJson, SimpleFixtureDo.class);
+    assertFalse(result.isChanged());
+    assertEquals(fixture, result.getDataObject());
+  }
+
+  /**
+   * Tests that the DO is not changed because the contribution to be removed still exists in the code.
```

Comment:
naming convention for plain text (java doc, comments, documentation: Use "data object", not "DO". Single exception: "DO entity" is fine.